traffic_ops/app/lib/API/DeliveryService/KeysUrlSig.pm

```diff
@@ -69,14 +86,17 @@ sub copy_url_sig_keys {
 	else {
 		return $self->alert("Delivery Service to copy from '$copy_from_xml_id' does not exist.");
 	}
+	if (!$tenant_utils->is_ds_resource_accessible($tenants_data, $copy_rs->tenant_id)) {
+		return $self->forbidden("Forbidden. Source delivery-service tenant is not available to the user.");
+	}
 	my $copy_config_file = $self->url_sig_config_file_name($copy_from_xml_id);
 
 	my $helper = new Utils::Helper( { mojo => $self } );
 	my $url_sig_key_values_json;
 
 	#verify we can copy keys out
 	if ( $helper->is_valid_delivery_service($copy_ds_id) ) {
-		if ( $is_admin || $helper->is_delivery_service_assigned($copy_ds_id) ) {
+		if ( $is_admin || $helper->is_delivery_service_assigned($copy_ds_id) || $tenant_utils->use_tenancy()) {
```

Comment:
Correct, the "|| use_tenancy" in this line come only to disable the legacy tests in case we use tenancy.
If the tenancy is on, we will not even get to this line (we will exit on line 90)